docs/write_test/extension.md

```diff
@@ -177,6 +179,16 @@ implement common and shareable logic. Learn more from `SerialConsole` in
     serial_console.check_panic(saved_path=case_path, stage="reboot")
     ```
 
+### Combinator
+
+The base class is [combinator.py](../lisa/combinator.py). The full matrix
```

Comment:
IMO if we are putting in links to code in documentation, we are failing completely at documenting. Why the need?

Reply:
Well, I think docs in write_test folder are all for test case developers, who differentiate from users/test runners by that they actually need to know technical things like our code examples or program design, and plus extensions are meant to be modified/extended, so it's necessary to provide all the links to code for reference. On the other hand, I suppose a detailed walkthrough of each extension component helps more, but I am not sure if it is necessary, or we have time to do that. Let me know your thoughts.

Reply:
Yeah, no worries if links are a complement. What worries me is just one sentence documenting something comprised of lots of Python code + a link.

Reply:
That's a good point. I think I will work on that.

Reply:
It's better than nothing under current heavy development work. We will improve and add more content based on user feedbacks continually.